src/supremm/xdmodaccount.py

```diff
@@ -31,14 +24,14 @@ def __init__(self, resource_id, config):
                 SELECT
                     jf.`job_id` AS `job_id`,
                     jf.`resource_id` AS `resource_id`,
-                    COALESCE(jf.`local_job_id_raw`, jf.`local_jobid`) AS `local_job_id`,
-            """
-            if job_uniq_mode == 'local_job_id':
-                self._query += "COALESCE(jf.`local_job_id_raw`, jf.`local_jobid`) as `job_uniq_id`, "
-            else:
-                self._query += "IF(jf.`local_job_array_index` = -1, jf.`local_jobid`, CONCAT(jf.`local_jobid`, '_', jf.`local_job_array_index`)) as job_uniq_id, "
-
-            self._query += """
+                    CASE
+                      WHEN jf.`local_job_id_raw` IS NULL THEN IF(jf.`local_job_array_index` = -1, jf.`local_jobid`, CONCAT(jf.`local_jobid`, '_', jf.`local_job_array_index`))
+                      WHEN sj.`source_format` = 'slurm' THEN jf.`local_job_id_raw`
+                      ELSE IF(jf.`local_job_array_index` = -1, jf.`local_jobid`, CONCAT(jf.`local_jobid`, '_', jf.`local_job_array_index`))
+                    END AS job_uniq_id,
```

Comment:
Does `job_uniq_id` need to be \``job_uniq_id`\` (with backticks)? Or is that a bit pedantic? Just thinking if it should match the syntax for the other aliases.